node/node.go

```diff
@@ -193,7 +202,7 @@ func BootstrapState(ctx context.Context, config *cfg.Config, dbProvider cfg.DBPr
 		return fmt.Errorf("state not empty, trying to initialize non empty state")
 	}
 
-	genState, _, err := LoadStateFromDBOrGenesisDocProvider(stateDB, DefaultGenesisDocProviderFunc(config))
+	genState, _, err := LoadStateFromDBOrGenesisDocProvider(stateDB, genProvider)
```

Comment:
this changes the semantics a bit, because you're saying that the `config` used in `DefaultGenesisDocProviderFunc(config)` when invoking this function is the same `config` (2nd parameter) in the `BootstrapStateWithGenProvider` function. In some respect I think in the end the original code having to pass `config` to DefaultGenesisDocProviderFunc is weird, probably it should have been a receiver in config. Anyway, I think this change is fine, I don't see any other use cases or any issues that this might cause a problem (I don't think anyone will invoke `BootstrapStateWithGenProvider` with two config :-)